### ISSUE TITLE
Cant install with laravel 8

### ISSUE THREAD

**Issue by soncemvo** [bringyourownideas/laravel-backblaze#6]
Laravel 8 now require guzzle 7.0.1 so composer fails:

```
Problem 1
    - cwhite92/b2-sdk-php[v1.2.0, ..., v1.3.0] require guzzlehttp/guzzle ^6.1 -> found guzzlehttp/guzzle[6.1.0, ..., 6.5.x-dev] but it conflicts with your root composer.json require (^7.0.1).
    - gliterd/backblaze-b2[dev-analysis-qoj29N, dev-analysis-8w1029, 0.0.1, ..., 0.0.3, 1.0.0, ..., 1.1.1] require guzzlehttp/guzzle ^6.1 -> found guzzlehttp/guzzle[6.1.0, ..., 6.5.x-dev] but it conflicts with your root composer.json require (^7.0.1).
    - gliterd/backblaze-b2[dev-feature/refactor, dev-release/1.2.0, dev-analysis-L3oKZl, 1.2.0, ..., 1.3.1] require php ^7.1.8 -> your php version (8.0.1) does not satisfy that requirement.
    - mhetreramesh/flysystem-backblaze[dev-release/1.5.0, dev-analysis-XZbavP, dev-analysis-q5p9Oy, 1.5.0, ..., 1.5.2] require php ^7.1.8 -> your php version (8.0.1) does not satisfy that requirement.
    - gliterd/backblaze-b2[dev-feature/update-dependencies, dev-master, dev-develop, dev-analysis-4xy9PE, 1.4.0, ..., 1.4.2] require php ^7.2 -> your php version (8.0.1) does not satisfy that requirement.
    - bringyourownideas/laravel-backblaze 2.0.0 requires mhetreramesh/flysystem-backblaze * -> satisfiable by mhetreramesh/flysystem-backblaze[dev-feature/Directory_Functions, dev-release/1.5.0, dev-scrutinizer-patch-1, dev-analysis-XZbavP, dev-analysis-q5p9Oy, 1.0.0, ..., 1.5.2].
    - mhetreramesh/flysystem-backblaze[1.1.0, ..., 1.1.4] require gliterd/backblaze-b2 * -> satisfiable by gliterd/backblaze-b2[dev-feature/update-dependencies, dev-feature/refactor, dev-master, dev-release/1.2.0, dev-develop, dev-analysis-L3oKZl, dev-analysis-qoj29N, dev-analysis-4xy9PE, dev-analysis-8w1029, 0.0.1, 0.0.2, 0.0.3, 1.0.0, ..., 1.4.2].
    - mhetreramesh/flysystem-backblaze[dev-feature/Directory_Functions, dev-scrutinizer-patch-1, 1.0.0, ..., 1.0.10] require cwhite92/b2-sdk-php ^1.2 -> satisfiable by cwhite92/b2-sdk-php[v1.2.0, v1.2.1, v1.3.0].
    - Root composer.json requires bringyourownideas/laravel-backblaze ^2.0 -> satisfiable by bringyourownideas/laravel-backblaze[2.0.0].
```
its not directly your problem, just letting you know

**Comment by spekulatius** [comment]
Ah okay, thanks for the heads up! 

As I'm doing my first project with Laravel 8 now I'll have to deal with it shortly too.

**Comment by spekulatius** [comment]
The dependency isn't looking overly active: https://github.com/cwhite92/b2-sdk-php 

This one https://github.com/zaxbux/b2-sdk-php might be worthy to look into - he might be more interested in adding guzzle 7 support

**Comment by slakbal** [comment]
Tip: From Laravel 8 if you use the built in http client then the Guzzle dependency can be dropped? Ah but I see that it is a dependency of a dependency.

This package has already Guzzle7 support: https://github.com/obregonco/backblaze-b2 maybe a good choice also.

**Comment by spekulatius** [comment]
Yeah, guzzle is a dependency of a dependency. I got to check if https://github.com/obregonco/backblaze-b2 works - looks promising

**Comment by slakbal** [comment]
Hmmmm, though I'd share this with you. Just tried something and it works like a charm. Since Backblaze says it has a S3 compatible storage API, I tried the following on the latest Laravel 8 with php8:

- Installed the Laravel recommended driver: https://laravel.com/docs/8.x/filesystem#composer-packages
- In my composer file I added the following dependency: 

`"league/flysystem-aws-s3-v3": "~1.0"`

- Then in my .env file added the following:

AWS_ACCESS_KEY_ID=xxxxxxxxxxxxxxxxxxxxx
AWS_SECRET_ACCESS_KEY=yyyyyyyyyyyyyyyyyyyyyy
AWS_DEFAULT_REGION=eu-central-003
AWS_BUCKET=TestBucket
AWS_ENDPOINT=https://s3.eu-central-003.backblazeb2.com

Note that I've added the "AWS_ENDPOINT" variable to the existing .env settings for AWS. All of the above mentioned information can be retrieved when creating a bucket and API key on backblaze's site.

- And then just use s3 as a disk e.g.: 

`Storage::disk('s3')->put('test-upload/test.txt', $contents)`

And BOOM it works! Just like as if you are using S3. I just figured if they say it is an S3 compatible API, then S3 adaptors should work.

Might not be optimal for all use-cases but for working with Laravel's storage and disks it seems to be honky dorky!

**Comment by soncemvo** [comment]
Not working for me, always got ACL error
`Error executing "PutObject" on "https://***.s3.us-west-002.backblazeb2.com/testb2-new.txt"; AWS HTTP error: Client error: `PUT https://***.s3.us-west-002.backblazeb2.com/testb2-new.txt` resulted in a `400 Bad Request` response: <?xml version="1.0" encoding="UTF-8" standalone="yes"?> <Error> <Code>InvalidArgument</Code> <Message>Unsupporte (truncated...) InvalidArgument (client): Unsupported value for canned acl 'private' - <?xml version="1.0" encoding="UTF-8" standalone="yes"?> <Error> <Code>InvalidArgument</Code> <Message>Unsupported value for canned acl 'private'</Message> </Error>`

so i compile all abandoned repos in single one, and fix some files for guzzle 7

**Comment by soncemvo** [comment]
> https://github.com/obregonco/backblaze-b2

there is some problems with this repo too

**Comment by spekulatius** [comment]
> Hmmmm, though I'd share this with you. Just tried something and it works like a charm. Since Backblaze says it has a S3 compatible storage API, I tried the following on the latest Laravel 8 with php8:
> 
>     * Installed the Laravel recommended driver: https://laravel.com/docs/8.x/filesystem#composer-packages
> 
>     * In my composer file I added the following dependency:
> 
> 
> `"league/flysystem-aws-s3-v3":`
> 
>     * Then in my .env file added the following:
> 
> 
> AWS_ACCESS_KEY_ID=xxxxxxxxxxxxxxxxxxxxx
> AWS_SECRET_ACCESS_KEY=yyyyyyyyyyyyyyyyyyyyyy
> AWS_DEFAULT_REGION=eu-central-003
> AWS_BUCKET=TestBucket
> AWS_ENDPOINT=https://s3.eu-central-003.backblazeb2.com
> 
> Note that I've added the "AWS_ENDPOINT" variable to the existing .env settings for AWS. All of the above mentioned information can be retrieved when creating a bucket and API key on backblaze's site.
> 
>     * And then just use s3 as a disk e.g.:
> 
> 
> `Storage::disk('s3')->put('test-upload/test.txt', $contents)`
> 
> And BOOM it works! Just like as if you are using S3. I just figured if they say it is an S3 compatible API, then S3 adaptors should work.
> 
> Might not be optimal for all use-cases but for working with Laravel's storage and disks it seems to be honky dorky!

That's a good point - at the end B2 should be compatible. I'll try this out for my project and let you know if it works for me.

**Comment by soncemvo** [comment]
s3 compatible api works only with master keys, as far as i know

**Comment by spekulatius** [comment]
Yeah, same for the current package - the application keys never worked.

**Comment by soncemvo** [comment]
u can use my repo if u want. i made application keys works here https://github.com/soncemvo/laravel-b2

**Comment by spekulatius** [comment]
> u can use my repo if u want. i made application keys works here https://github.com/soncemvo/laravel-b2

to achieve what exactly? I was thinking of going with the S3-compatible approach by @slakbal

**Comment by slakbal** [comment]
@soncemvo @spekulatius  I see I forgot to include the version on my composer entry. Updated my previous comment to include the following, but also included the full composer dependencies. As mentioned latest Laravel 8 and php8.

```json
{
        "php": "^7.3|^8.0",
        "fideloper/proxy": "^4.4",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "laravel/framework": "^8.12",
        "laravel/tinker": "^2.5",
        "league/flysystem-aws-s3-v3": "~1.0"
}
``` 

See docs: https://laravel.com/docs/8.x/filesystem#composer-packages

- the config is with "Application Keys" and **NOT**  "Master Keys". Make sure you give the application key all permissions for the respective bucket when creating it.
- AWS_DEFAULT_REGION=eu-central-003  ..... (My region is `eu-central-003`, which I extracted from the endpoint for the bucket for which the Application Key was generated:  s3. `eu-central-003` .backblazeb2.com Make sure you use the full value after s3. and up to .backblazeb2.com. 
- AWS_ENDPOINT=https://s3.eu-central-003.backblazeb2.com ..... and add **https://** to the end-point value.
- AWS_BUCKET=TestBucket  .... make sure the name is exactly the same as the name of the BackBlaze bucket.

https://www.backblaze.com/b2/docs/s3_compatible_api.html

Hope it's helps.

**Comment by spekulatius** [comment]
Awesome, thank you @slakbal! I'll try it out over the weekend and let you know if I got it working

**Comment by soncemvo** [comment]
@slakbal still no success with s3 
![image](https://user-images.githubusercontent.com/862204/106268597-652deb00-623c-11eb-9188-dec538ded159.png)
![image](https://user-images.githubusercontent.com/862204/106268691-7d9e0580-623c-11eb-9d85-c8520e833c47.png)

and this is full public bucket with all rights

**Comment by slakbal** [comment]
@soncemvo don't know what to tell you. I don't have access to your code so cant help you. I'll see if I can upload for you a quick project when I get some time. But let's also hear from @spekulatius if he was able to try it out also.

**Comment by spekulatius** [comment]
Hey @slakbal 

I've tested the s3-compatiblity and it works like a charm. Much easier - thanks for hinting to this. I think it makes more sense to suggest this approach instead of maintaining another package.

Cheers,
Peter

**Comment by spekulatius** [comment]
Hey @slakbal 

did you notice any issues while cleaning up backups? I seem to have the old backups remaining and an empty file being added?

Cheers,
Peter

**Comment by jiehui555** [comment]
I also met the same problem and found a solution.
In the `filesystem.php` configuation file, add an `options` item, and set the value of the `ACL` to an empty string.

```php
's3' => [
    'driver'                  => 's3',
    'key'                     => env('AWS_ACCESS_KEY_ID'),
    'secret'                  => env('AWS_SECRET_ACCESS_KEY'),
    'region'                  => env('AWS_DEFAULT_REGION'),
    'bucket'                  => env('AWS_BUCKET'),
    'url'                     => env('AWS_URL'),
    'endpoint'                => env('AWS_ENDPOINT'),
    'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
    'options'                 => [
        'ACL' => '',
    ]
],
```

**Comment by spekulatius** [comment]
Hello @hui-ho,

I've tried the settings but my old backups still won't get cleaned up. Any idea why?

Cheers,
Peter
